practice/031/ex.ml

```diff
@@ -0,0 +1,15 @@
+open OUnit2
+
+module type Testable = sig
+  val is_prime : int -> bool
+end
+
+module Make(Tested: Testable) : sig val v : test end = struct
+  let v = "is_prime" >::: [
+    "nil" >:: (fun _ -> assert_equal false (Tested.(not (is_prime 7))));
+    "cons" >:: (fun _ -> assert_equal true (Tested.is_prime 7));
```

Comment:
* Remove test `nil`.
* Define a list of primes such as: 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97
* Compute the list of tests for all values from 0 to 100 from the list above

Reply:
Kindly check if what I have done if what expect me to do, if not, please point me in the right direction, thanks.

Reply:
This is better, but it does not compile because `primes` has type `int list` and the anonymous function you pass to `List.map` has type `(int * bool) list`. However, this is indeed what I want.

You need to build the `(int * bool) list` from the `primes` using this function:
```ocaml
  let rec flag_primes n = function
  | [] -> []
  | p :: u -> if n = p then
                (n, true) :: flag_primes (n + 1) u
              else
                (n, false) :: flag_primes (n + 1) (p :: u)
```

Reply:
Hi @cuihtlauac thank you for your feedback. This is what I have been able to come up with, is this what you expect?